app/shared/JSONRequest.js

```diff
@@ -6,6 +6,18 @@ import rison from 'rison-node';
 
 let cookie;
 
+class FetchResponseError extends Error {
+  // eslint-disable-next-line no-shadow
+  constructor(message, { json, status, cookie, headers } = {}) {
+    super(message);
+    this.name = 'FetchResponseError';
+    this.json = json;
+    this.status = status;
+    this.cookie = cookie;
```

Comment:
@LaszloKecskes Do you think it's a good idea to include the cookie and headers in the response? cc @daneryl @RafaPolit

Reply:
Well, actually that data is not being logged into the logs nor sent on the error response, so that's fine.